cpp/src/io/functions.cpp

```diff
@@ -68,6 +68,8 @@ compression_type infer_compression_type(compression_type compression, source_inf
   if (ext == "gz") { return compression_type::GZIP; }
   if (ext == "zip") { return compression_type::ZIP; }
   if (ext == "bz2") { return compression_type::BZIP2; }
+  if (ext == "zstd") { return compression_type::ZSTD; }
```

Comment:
We should optimize all our code paths for exiting early based on the distribution of most common compression types. If zstd and snappy are most popular, put them first. This function is cheap but other functions like the AUTO fallthrough logic above might not be, and a consistent ordering across all of our I/O code would be nice.